applications/StructuralMechanicsApplication/custom_constitutive/serial_parallel_rule_of_mixtures_law.cpp

```diff
@@ -0,0 +1,1013 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:         BSD License
+//                   license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Alejandro Cornejo 
+//                   Vicente Mataix
+//                   Fernando Rastellini
+//  Collaborator:    Lucia Barbu
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "utilities/math_utils.h"
+#include "structural_mechanics_application_variables.h"
+#include "serial_parallel_rule_of_mixtures_law.h"
+#include "custom_utilities/tangent_operator_calculator_utility.h"
+
+
+namespace Kratos
+{
+ConstitutiveLaw::Pointer SerialParallelRuleOfMixturesLaw::Create(Kratos::Parameters NewParameters) const
+{
+    const double fiber_volumetric_participation = NewParameters["combination_factors"][1].GetDouble();
+    const int voigt_size = 6;
```

Comment:
This should depend of the input laws!!!

Reply:
this formulation is developed in 3D... Another problem is that here the onput laws do not exist right?

Reply:
I would use the information from the input laws